File: TemplateNetCore-main/MercanciaSegura.DOM/Modelos/Cliente/BeneficiarioPreferente.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MercanciaSegura.DOM.Modelos.Cliente
{
    [Table("Beneficiario_Preferente")]
    public class BeneficiarioPreferente
    {
        [Key]
        [Column("Beneficiario_Preferente_ID")]
        public int BeneficiarioPreferenteId { get; set; }

        [Column("Nombre")]
        [MaxLength(100)]
        public string? Nombre { get; set; }

        [Column("Domicilio")]
        [MaxLength(200)]
        public string? Domicilio { get; set; }

        [Column("RFC")]
        [MaxLength(13)]
        public string? RFC { get; set; }

        [Column("Clave")]
        [MaxLength(10)]
        public string? Clave { get; set; }

        [Column("Tipo_Persona_ID")]
        public int? TipoPersonaId { get; set; }

        [Column("Apellido_Paterno")]
        [MaxLength(60)]
        public string? ApellidoPaterno { get; set; }

        [Column("Apellido_Materno")]
        [MaxLength(60)]
        public string? ApellidoMaterno { get; set; }

        [Column("Nombre_Completo")]
        [MaxLength(120)]
        public string? NombreCompleto { get; set; }

        [Column("RfcGenerico_ID")]
        public int? RfcGenericoId { get; set; }

        [Column("Calle")]
        [MaxLength(120)]
        public string? Calle { get; set; }

        [Column("Numero_Int")]
        [MaxLength(10)]
        public string? NumeroInt { get; set; }

        [Column("Numero_Ext")]
        [MaxLength(10)]
        public string? NumeroExt { get; set; }

        [Column("Poblacion")]
        [MaxLength(80)]
        public string? Poblacion { get; set; }

        [Column("Colonia")]
        [MaxLength(60)]
        public string? Colonia { get; set; }

        [Column("CP")]
        [StringLength(5)]
        public string? Cp { get; set; }

        [Column("Pais")]
        [MaxLength(50)]
        public string? Pais { get; set; }

        [Column("Nacionalidad")]
        [MaxLength(30)]
        public string? Nacionalidad { get; set; }

        [Column("Fecha_Actualizacion")]
        public DateTime? FechaActualizacion { get; set; }

        [Column("Fecha_Registro")]
        public DateTime FechaRegistro { get; set; } = DateTime.Now;

        [ForeignKey(nameof(TipoPersonaId))]
        public TipoPersona? TipoPersona { get; set; }

        [ForeignKey(nameof(RfcGenericoId))]
        public RfcGenerico? RfcGenerico { get; set; }
    }
}

File: TemplateNetCore-main/WebAdmin/Modelos/BeneficiarioPreferente.vb
Imports Newtonsoft.Json

Namespace MercanciaSegura.DOM.Modelos

    Public Class BeneficiarioPreferente

        Public Property BeneficiarioPreferenteId As Integer

        Public Property RFC As String
        Public Property ClienteId As Integer?
        Public Property Clave As String
        Public Property TipoPersonaId As Integer?
        Public Property ApellidoPaterno As String
        Public Property ApellidoMaterno As String
        Public Property Nombre As String
        Public Property NombreCompleto As String
        Public Property RfcGenericoId As Integer?
        Public Property rfcGenerico As String
        Public Property Calle As String
        Public Property NumeroInt As String
        Public Property NumeroExt As String
        Public Property Pais As String
        Public Property Poblacion As String
        Public Property Colonia As String
        Public Property Cp As String
        Public Property Nacionalidad As String
        Public Property FechaActualizacion As Date?
        Public Property FechaRegistro As Date = Date.Now

        Public ReadOnly Property RFCAMostrar As String
            Get
                If Not String.IsNullOrEmpty(Me.RFC) Then
                    Return Me.RFC
                End If

                If Me.RfcGenericoId.HasValue Then
                    Dim listaRFCGenericos As List(Of rfcGenerico) = TryCast(HttpContext.Current.Session("ListaRFCGenericos"), List(Of rfcGenerico))
                    If listaRFCGenericos IsNot Nothing Then
                        Dim seleccionado = listaRFCGenericos.FirstOrDefault(Function(x) x.RfcGenericoId = Me.RfcGenericoId.Value)
                        If seleccionado IsNot Nothing Then
                            Return seleccionado.Tipo
                        End If
                    End If
                End If

                Return String.Empty
            End Get
        End Property
    End Class
End Namespace